Sources/ManagedAppConfigLib/AppConfig.swift

```diff
@@ -0,0 +1,86 @@
+/*
+    SPDX-License-Identifier: MIT
+    Copyright (c) 2022 Jamf Open Source Community
+*/
+
+import Foundation
+import SwiftUI
+
+/// A property wrapper type that reflects a value from Managed App Config (via `UserDefaults`) and
+/// invalidates a SwiftUI view on a change in value in that Managed App Config.
+@available(macOS 11, iOS 13.0, tvOS 13.0, *)
+@propertyWrapper public struct AppConfig<Value>: DynamicProperty {
+    // Very simple listener that observes AppConfig changes, and has a local copy of the AppConfig's value.
+    private final class Listener<Value>: ObservableObject {
+        var subscriber: NSObjectProtocol?
+        var value: Value? {
+            willSet {
+                self.objectWillChange.send()
+            }
+        }
+
+        func listenTo(store: UserDefaults, key: String, defaultValue: Value) {
+            if subscriber == nil {
+                AppConfigService.shared.use(userDefaults: store)
+                value = AppConfigService.shared.value(for: key, store) ?? defaultValue
+                subscriber = NotificationCenter.default.addObserver(forName: .appConfigMayHaveChanged,
+                                                    object: store, queue: nil) { [weak self] _ in
+                    guard let self = self else { return }
```

Comment:
is `self` an optional?

Reply:
Yes, because of the `[weak self] _ in` on the previous line.